backend-express: Remove unused '_startMemory' variable (TS6133)

## Problem

A TypeScript compilation error (TS6133) blocks the full monorepo build. The variable `_startMemory` is declared on line 230 of the stress-testing test file but is never used, causing the build to fail.

## Error Details

**Error Message:**
```
backend-express build: src/routes/__tests__/stress-testing.test.ts(230,11): error TS6133: '_startMemory' is declared but its value is never read.
backend-express build: Failed
```

**Location:** `backend-express/src/routes/__tests__/stress-testing.test.ts:230`

**Error Type:** Unused variable (TypeScript strict mode)

## Impact

- Running `pnpm build` fails at the backend-express package compilation stage
- Blocks CI/CD pipelines
- Frontend and tests pass independently, but full monorepo build cannot complete
- **Severity:** Medium (not critical, but prevents full deployment builds)

## Root Cause

Line 230 declares:
```typescript
const _startMemory = getHeapUsedMB();
```

However, the variable is never referenced elsewhere in the test. The test uses a `measurements` array to track memory samples at regular intervals (every 6000 events) and compares `firstMemory` and `lastMemory` from the measurements array rather than using the captured start value.

## Solution Options

1. **Remove the unused variable** (if start memory capture is not needed)
   - Simply delete line 230
   - The test already captures initial memory via `measurements[0].memory`

2. **Use the variable** (if it's needed for documentation/comparison)
   - Use `_startMemory` for an assertion or console logging
   - Document why initial memory baseline is captured

## Verification

After fix, verify with:
```bash
pnpm build
```

## Context

- **Found During:** PR #197 code review (Issue #30: Optimistic Updates)
- **Pre-existing Issue:** Not related to Issue #30 changes
- **Discovered:** April 17, 2026

## Related Issues

- Issue #30 (Optimistic Updates - code review context)
- PR #197

## Test Details

This is part of the 'Event Bus Stress Testing - Suite 2: Memory Leak Detection' test suite, which validates that the event bus doesn't have memory leaks over sustained load (30,000 events over 300 simulated seconds).

--- a/backend-express/src/routes/__tests__/stress-testing.test.ts
+++ b/backend-express/src/routes/__tests__/stress-testing.test.ts
@@ -227,7 +227,6 @@ describe('Event Bus Stress Testing - Suite 1: High-Throughput', () => {
 describe('Event Bus Stress Testing - Suite 2: Memory Leak Detection', () => {
   it('2.1: event bus memory stays stable over sustained load', async () => {
     const measurements: Array<{ time: number; memory: number }> = [];
-    const _startMemory = getHeapUsedMB();
     const metricsCollector = new EventBusMetricsCollector();
 
     // Run for 300 simulated seconds (5-minute equivalent)

